### ISSUE TITLE
Support redshift 'ignore nulls' clause

### ISSUE THREAD

**Issue by nolanbconaway** [sqlfluff/sqlfluff#1379]
We've been using postgres as a standin for redshift, and this is the first time i've seen some redshift-specific syntax get in the way (credit to @WittierDinosaur).

I adapted this [query](https://online.sqlfluff.com/fluffed?sql=H4sIAENlpWAC_2WOSwqDQBBE9wHv0EuF3EnmU8rA2C09PUpuH40kDqSWrx5UFWQE6x50ZANXFHOGZwvgrLSA3fIVpqTFxs3liv7XUZpZFMQ15zJcomzQfnVqyZIw-Vczdhn_EY3QWz1vUEQJt6-yF_KwHWCq7KVyRKRVERATz-Q4NnySnGU_-ECuUJA8nne7x6SykDmf8QFvWEDYMRMBAAA%3D&dialect=postgres) from the [redshift docs](https://docs.aws.amazon.com/redshift/latest/dg/r_WF_first_value.html):

```sql
select
    venuestate,
    venueseats,
    venuename,
    first_value(venuename ignore nulls)
    over(partition by venuestate
                      order by venueseats desc
        rows between unbounded preceding and unbounded following) as col_name
from table_name
```

I suppose this belongs on the laundry list of reasons to one day support a dedicated redshift dialect.

**Comment by WittierDinosaur** [comment]
To add to this discussion, I think we should consider adding redshift support as primarily a copy of Postgres. Redshift is based on Postgres, so extensions should (hopefully) be rare. Overwriting functionality in the redshift dialect will likely be easier to maintain than basing it on ANSI, and writing any Postgres stuff out twice.

**Comment by WittierDinosaur** [comment]
So, we may need a redshift dialect, but it looks like not for this. The reason this didn't parse was a couple of bugs in the ANSI dialect. I've beefed up the FrameClauseSegment to meet the spec in your docs. and moved IGNORE/ RESPECT NULLS out of the PostFunctionGrammar and into the FunctionContentsGrammar, and it seems to work. I've added your example to the tests in Postgres. PR #1381 contains the fix
